frontend/src/App.js

```diff
@@ -110,7 +127,7 @@ export default function App() {
             )}
 
             {user === "owner" && <Owner />}
-            {developerMode && user === "customer" && (
+            {developerMode && user === "customer" && currentView === "scan" && (
               <Button onPress={() => (state.user = "owner")}>Owner view</Button>
             )}
           </ScrollView>
```

Comment:
The code adds a new function called `setRestaurantDetails` which takes a restaurant name or hostname and retrieves the corresponding restaurant information through an API. The function updates the state of the component with the `restaurantId`, `restaurantName`, and `products`. There are no major bug risks identified, but some suggestions for improvement are:

- Remove commented-out code because it adds confusion to future readers.
- Consider removing repeated alert messages in the `setRestaurantDetails` function and throwing an error instead. 
- Check whether the API call in `setRestaurantDetails` is successful and handle errors appropriately by catching any exceptions.
- Change `alert` calls to a more user-friendly approach (e.g., displaying a message on the screen instead of using an `alert`).
- Consider breaking up long conditional blocks into separate functions for readability.
- Use a constant to simplify the boolean expression and improve readability.
- Consider using `switch` statements instead of nested `if` statements to improve readability.

Finally, there's an interesting change constructed on lines 123-124 where the variable `currentView` is added to the condition for the `developerMode`. This is suggested for cleaner code that correctly displays the Owner button only in the customer view.